recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,154 @@
+import os
+from conans import ConanFile, CMake, tools
+
+class SociConan(ConanFile):
+    name = "soci"
+    homepage = "https://github.com/SOCI/soci"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "The C++ Database Access Library "
+    topics = ("C++", "database-library", "oracle", "postgresql", "mysql", "odbc", "db2", "firebird", "sqlite3", "boost" )
+    license = "Boost"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    options = {
+        "fPIC":       [True, False],
+        "static":     [True, False],
+        "cxx11":      [True, False],
+        "empty":      [True, False],
+        "shared":     [True, False],
+        "sqlite3":    [True, False],
+        "db2":        [True, False],
+        "odbc":       [True, False],
+        "oracle":     [True, False],
+        "firebird":   [True, False],
+        "mysql":      [True, False],
+        "postgresql": [True, False],
+        "boost":      [True, False],
+        "valgrind":   [True, False]
+    }
+
+    default_options = {
+        "fPIC":       True,
+        "static":     True,
+        "cxx11":      True,
+        "empty":      True,
+        "shared":     False,
+        "sqlite3":    False,
+        "db2":        False,
+        "odbc":       False,
+        "oracle":     False,
+        "firebird":   False,
+        "mysql":      False,
+        "postgresql": False,
+        "boost":      False,
+        "valgrind":   False
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        prefix  = "Dependencies for "
+        message = " not configured in this conan package, some features will be disabled."
```

Comment:
Any option affects the package ID, so if a dependency is listed but not yet available, you should raise a ConanInvalidConfiration instead. A warning message is not enough.